components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/CodeTokenResponseValidator.java

```diff
@@ -0,0 +1,91 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.common;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.oltu.oauth2.as.validator.TokenValidator;
+import org.apache.oltu.oauth2.common.OAuth;
+import org.apache.oltu.oauth2.common.error.OAuthError;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuth20Params.NONCE;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuth20Params.SCOPE;
+
+/**
+ * Validator for hybrid flow code token requests
+ * identity.xml
+ */
+public class CodeTokenResponseValidator extends TokenValidator {
+
+    public CodeTokenResponseValidator() {
+    }
+
+    @Override
+    public void validateRequiredParameters(HttpServletRequest request) throws OAuthProblemException {
+
+        super.validateRequiredParameters(request);
+
+        String nonce = request.getParameter(NONCE);
+        if (StringUtils.isBlank(nonce)) {
+            throw OAuthProblemException.error(OAuthError.TokenResponse.INVALID_REQUEST)
+                    .description("\'response_type\' for \'hybrid flow\'; but \'nonce\' parameter not found");
+        }
+
+        // For code token response type, the scope parameter should contain 'openid' as one of the scopes.
+        String openIdScope = request.getParameter(SCOPE);
+        if (StringUtils.isBlank(openIdScope) || !isContainOIDCScope(openIdScope)) {
+            throw OAuthProblemException.error(OAuthError.TokenResponse.INVALID_REQUEST)
+                    .description("\'response_type\' for \'hybrid flow\'; but \'openid\' scope not found.");
+        }
+
+    }
+
+    @Override
+    public void validateMethod(HttpServletRequest request) throws OAuthProblemException {
+
+        String method = request.getMethod();
+        if (!OAuth.HttpMethod.GET.equals(method) && !OAuth.HttpMethod.POST.equals(method)) {
+            throw OAuthProblemException.error(OAuthError.CodeResponse.INVALID_REQUEST)
+                    .description("Method not correct.");
+        }
+    }
+
+    @Override
+    public void validateContentType(HttpServletRequest request) throws OAuthProblemException {
+    }
+
+    /**
+     * Method to check whether the scope parameter string contains 'openid' as a scope.
+     *
+     * @param scope
+     * @return
+     */
+    private static boolean isContainOIDCScope(String scope) {
```

Comment:
no need of adding a new method for this. Re-use OAuth2Util.isOIDCAuthzRequest

Reply:
But the existing function take different arguments.